File: examples/fsharp/ShapeService.fs
module JsonTag.FSharp.Example.ShapeService

open System
open JsonTag.FSharp.Example

let private getRandomShapeEdge () =
    Random.Shared.Next(80, 121)

let generateRandomShape () =
    match Random.Shared.Next(3) with
    | 0 -> Circle(diameter = getRandomShapeEdge ())
    | 1 -> Rectangle(width = getRandomShapeEdge (), height = getRandomShapeEdge ())
    | _ -> Triangle(edgeA = getRandomShapeEdge (), edgeB = getRandomShapeEdge (), edgeC = getRandomShapeEdge ())

File: examples/csharp/ShapeService.cs
namespace JsonTag.Example;

public static class ShapeService
{
    public static Shape GenerateRandomShape()
    {
        var variant = Random.Shared.Next(3);
        return variant switch
        {
            0 => new Circle { Diameter = GetRandomShapeEdge() },
            1 => new Rectangle { Width = GetRandomShapeEdge(), Height = GetRandomShapeEdge() },
            _ => new Triangle { EdgeA = GetRandomShapeEdge(), EdgeB = GetRandomShapeEdge(), EdgeC = GetRandomShapeEdge() },
        };
    }

    private static int GetRandomShapeEdge() => Random.Shared.Next(80, 121);
}
